cmd/edge-proxy/main.go

```diff
@@ -63,67 +64,69 @@ func main() {
 	flag.StringVar(&httpTunnelAddr, "http-tunnel-listen", "localhost:8888", "Listen address for HTTP (CONNECT) tunnel server")
 	flag.Parse()
 
+	proxyOnlyMode = false
 	if proxyURI == "" {
-		fmt.Printf("proxy-uri must be provided\n")
-
-		os.Exit(1)
-	}
-
-	proxyURIParsed, err := url.Parse(proxyURI)
-
-	if err != nil {
-		fmt.Printf("proxy-uri invalid: %s\n", err.Error())
-
-		os.Exit(1)
+		fmt.Printf("proxy-uri not provided so starting in proxy-only mode\n")
```

Comment:
The value of `proxyOnlyMode` doesn't seem to relate to the parameter `proxyURI` too much for me. Maybe we could have an extra command line parameter for client to set it up only instead of checking the emptiness of `proxyURI`. Except this, other changes look good to me.

Reply:
Left over from a refactor good catch.

Reply:
We actually did that at first but we took it out because we had to have a lot of verification logic in our launch scripts that overcomplicated things.